src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int SIZE_INCREASE_INDEX = 2;
+    private int capacityTable = 16;
+    private int threshold = (int) (capacityTable * DEFAULT_LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[capacityTable];
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        putInEmptyBox(key, value, findIndex(key));
+        putInNotEmptyBox(key, value, findIndex(key));
+    }
+
+    private void putInEmptyBox(K key, V value, int index) {
+        if (table[index] == null) {
+            table[findIndex(key)] = new Node<>(key, value, null);
+            size++;
```

Comment:
why do we have to call findIndex(key) here if we already have our index

Reply:
I've already got index in method put and no reason to use this again. My bad